### PR TITLE
feat: Category store integration

### PR DESCRIPTION
**Pull request by iamsivin**
# Pull Request Template

## Description

1. Adds the ability to create a category.
2. Now you can see all the categories and their count in the sidebar.
3. Adds the ability to see articles based on categories.

Fixes https://github.com/chatwoot/product/issues/517

## Type of change

- [x] New feature (non-breaking change which adds functionality)

## How Has This Been Tested?
https://www.loom.com/share/2322cc66ba6847a291d3352319d45b40

## Checklist:

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [ ] I have commented on my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes
- [ ] Any dependent changes have been merged and published in downstream modules

### CODE REVIEW

**Review by nithindavid — APPROVED** [review]

File: app/javascript/dashboard/components/layout/sidebarComponents/SecondaryNavItem.vue [hunk 1]
```diff
@@ -137,7 +148,25 @@ export default {
...
       return (
         this.$store.state.route.name === 'list_all_locale_articles' &&
-        this.menuItem.toStateName === 'all_locale_articles'
+        this.menuItem.toStateName === 'list_all_selectedPortalLocale_articles'
+      );
+    },
+    isMyArticles() {
+      return (
+        this.$store.state.route.name === 'list_mine_articles' &&
+        this.menuItem.toStateName === 'mine_articles'
```
Comment (pranavrajs) [inline app/javascript/dashboard/components/layout/sidebarComponents/SecondaryNavItem.vue:157]: Is it mine_articles or list_mine_articles? 

Also, this code seems to be repeated below. Can we create a method?

File: app/javascript/dashboard/routes/dashboard/helpcenter/pages/categories/ShowCategory.vue [hunk 2]
```diff
@@ -1,3 +1,135 @@
 <template>
```
Comment (muhsin-k) [inline app/javascript/dashboard/routes/dashboard/helpcenter/pages/categories/ShowCategory.vue:1]: @iamsivin Let's use [ListAllArticles](https://github.com/chatwoot/chatwoot/blob/develop/app/javascript/dashboard/routes/dashboard/helpcenter/pages/articles/ListAllArticles.vue) with category filter

Comment (iamsivin) [reply app/javascript/dashboard/routes/dashboard/helpcenter/pages/categories/ShowCategory.vue:1]: Done

**Comment by muhsin-k** [comment]
@iamsivin Please fix these alignment issues
Screenshot 2022-08-09 at 8 09 35 PM

File: app/javascript/dashboard/routes/dashboard/helpcenter/components/HelpCenterLayout.vue [hunk 3]
```diff
@@ -33,6 +34,12 @@
         :active-portal="selectedPortal"
         @close-popover="closePortalPopover"
       />
+      <add-category
```
Comment (muhsin-k) [inline app/javascript/dashboard/routes/dashboard/helpcenter/components/HelpCenterLayout.vue:37]: Use `<woot-modal>` here

File: app/javascript/dashboard/routes/dashboard/helpcenter/components/HelpCenterLayout.vue [hunk 4]
```diff
@@ -142,26 +157,15 @@ export default {
...
-          ],
+          children: this.categories.map(view => ({
+            id: view.id,
+            label: view.name,
+            count: view.meta.articles_count,
+            truncateLabel: true,
+            toState: frontendURL(
+              `accounts/${this.accountId}/portals/${this.selectedPortalSlug}/${view.locale}/categories/${view.slug}`
+            ),
+          })),
```
Comment (muhsin-k) [inline app/javascript/dashboard/routes/dashboard/helpcenter/components/HelpCenterLayout.vue:168]:
```suggestion
          children: this.categories.map(category => ({
            id: category.id,
            label: category.name,
            count: category.meta.articles_count,
            truncateLabel: true,
            toState: frontendURL(
              `accounts/${this.accountId}/portals/${this.selectedPortalSlug}/${category.locale}/categories/${category.slug}`
            ),
          })),
```

**Review by muhsin-k — APPROVED** [review]

**Pull request merged**